pkg/engine/mutate/mutation.go

```diff
@@ -77,7 +75,7 @@ func ForEach(name string, foreach kyvernov1.ForEachMutation, policyContext engin
 	if err != nil {
 		return NewErrorResponse("variable substitution failed", err)
 	}
-	patcher := NewPatcher(fe.GetPatchStrategicMerge(), fe.PatchesJSON6902)
+	patcher := NewPatcher(fe["patchStrategicMerge"], fe["patchesJson6902"].(string))
```

Comment:
Do we need a type assertion for "fe["patchesJson6902"].(string)"?

Can we add unit tests for this?

Reply:
We do need it because `NewPatcher` needs a string. But it's very safe because it's already a string when put in the map and variable substitution doesn't change type.

This is where it's put in the map:

```
patchesMap["patchesJson6902"] = fe.PatchesJSON6902
```

`fe.PatchesJSON6902` is already a string.

Reply:
Ok, let's cover it by unit tests.